skywalking-sniffer/skywalking-sdk-plugin/mongodb-plugin/src/main/java/com/a/eye/skywalking/plugin/mongodb/MongoDBMethodInterceptor.java

```diff
@@ -0,0 +1,139 @@
+import com.mongodb.operation.GroupOperation;
+import com.mongodb.operation.InsertOperation;
+import com.mongodb.operation.ListCollectionsOperation;
+import com.mongodb.operation.MapReduceToCollectionOperation;
+import com.mongodb.operation.MapReduceWithInlineResultsOperation;
+import com.mongodb.operation.MixedBulkWriteOperation;
+import com.mongodb.operation.UpdateOperation;
+
+/**
+ * {@link MongoDBMethodInterceptor} intercept method of {@link com.mongodb.Mongo#execute(ReadOperation, ReadPreference)}
+ * or {@link com.mongodb.Mongo#execute(WriteOperation)}. record the mongoDB host, operation name and the key of the operation.
+ *
+ * @author baiyang
+ */
+public class MongoDBMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+    /**
+     * The key name that MongoDB host in {@link EnhancedClassInstanceContext#context}.
+     */
+    protected static final String MONGODB_HOST = "MONGODB_HOST";
+
+    /**
+     * The key name that MongoDB port in {@link EnhancedClassInstanceContext#context}.
+     */
+    protected static final String MONGODB_PORT = "MONGODB_PORT";
+
+    private static final String MONGODB_COMPONENT = "MongoDB";
+    
+    @Override
+    public void beforeMethod(final EnhancedClassInstanceContext context,
+            final InstanceMethodInvokeContext interceptorContext, final MethodInterceptResult result) {
+        Object[] arguments = interceptorContext.allArguments();
+        OperationInfo operationInfo = this.getReadOperationInfo(arguments[0]);
+        Span span = ContextManager.createSpan("MongoDB/" + operationInfo.getMethodName());
+        Tags.COMPONENT.set(span, MONGODB_COMPONENT);
+        Tags.DB_TYPE.set(span, MONGODB_COMPONENT);
+        Tags.SPAN_KIND.set(span, Tags.SPAN_KIND_CLIENT);
+        Tags.SPAN_LAYER.asDB(span);
+        Tags.DB_STATEMENT.set(span, operationInfo.getMethodName() + " " + operationInfo.getFilter());
+    }
+
+    @Override
+    public Object afterMethod(EnhancedClassInstanceContext context, InstanceMethodInvokeContext interceptorContext,
+            Object ret) {
+        Span span = ContextManager.activeSpan();
+        Tags.PEER_HOST.set(span, context.get(MONGODB_HOST, String.class));
+        Tags.PEER_PORT.set(span, (Integer) context.get(MONGODB_PORT));
+        ContextManager.stopSpan();
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(Throwable t, EnhancedClassInstanceContext context,
+            InstanceMethodInvokeContext interceptorContext) {
+        ContextManager.activeSpan().log(t);
+    }
+
+    /**
+     *  Convert ReadOperation interface or WriteOperation interface to the implementation class.
+     *  Get the method name and filter info.
+     */
+    @SuppressWarnings("rawtypes")
+    private OperationInfo getReadOperationInfo(Object obj) {
+        if (obj instanceof CountOperation) {
+            return new OperationInfo(ReadMethod.COUNT.getName(), ((CountOperation) obj).getFilter().toString());
```

Comment:
@wu-sheng  Here is some snapshots about the span detail of Mongdb span.
1、Find method：
![find](https://cloud.githubusercontent.com/assets/20091216/25138744/e985574e-248d-11e7-9253-0ee9124c148d.jpeg)
2、Insert method:
![insert](https://cloud.githubusercontent.com/assets/20091216/25138753/eda9e772-248d-11e7-951a-9551a310df2e.jpeg)